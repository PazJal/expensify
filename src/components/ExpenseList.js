import React from 'react';
import {connect} from 'react-redux';
import FlipMove from 'react-flip-move';

import selectExpenses from './../selectors/expenses'
import ExpenseListItem from './ExpenseListItem';

const ExpenseList = (props) => (
  <div>
    <h1>ExpenseList</h1>
    <FlipMove>
    {props.expenses.map((expense) => {
      return (
        <ExpenseListItem key={expense.id} {...expense} />
      );
    })}
    </FlipMove>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);

