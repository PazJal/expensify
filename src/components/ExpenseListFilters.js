import React from 'react';
import {connect} from 'react-redux';
import {DateRangePicker} from 'react-dates';


import {setTextFilter , sortByAmount , sortByDate , setStartDate ,setEndDate} from './../actions/filters';

export class ExpenseListFilters extends React.Component {

  state = {
    calanderFocused: null
  };

  onDatesChange = ({startDate , endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calanderFocused) => {
    this.setState(() => ({calanderFocused}));
  };

  onSelectChange = (e) => {
    if(e.target.value === 'date') {
      this.props.sortByDate();
    } else if(e.target.value === 'amount'){
      this.props.sortByAmount();
    }
  }

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  }

  render() {
    return (
      <div>
        <input type="text" 
          value={this.props.filters.text} 
          onChange={this.onTextChange} 
          name="" 
          id=""
        />
        <select name="" id="" 
          value={this.props.filters.sortBy} 
          onChange={this.onSelectChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker 
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calanderFocused}
          onFocusChange={this.onFocusChange}
          startDateId={'1'}
          endDateId={'1'}
          numberOfMonths={1}
          isOutsideRange={() => (false)}
          showClearDates={true}
        />
    
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps , mapDispatchToProps)(ExpenseListFilters);