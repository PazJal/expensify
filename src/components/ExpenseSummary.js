import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/selectExpensesTotal';


export const ExpenseSummary = ({expenseCount , expenseTotal}) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpenseTotal = numeral(expenseTotal / 100).format('$0,0.00');
  
  return (
    <div>
      Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}
    </div>
  );

}
  

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses , state.filters);
  return {
    expenseCount: expenses.length,
    expenseTotal: selectExpenseTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);