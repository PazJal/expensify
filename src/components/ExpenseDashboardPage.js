import React from 'react';

import ExpenseListFilters from './ExpenseListFilters';
import ExpenseList from './ExpenseList';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseSummary />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;