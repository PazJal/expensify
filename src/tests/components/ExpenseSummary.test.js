import React from 'react';
import {shallow} from 'enzyme';

import {ExpenseSummary} from '../../components/ExpenseSummary';

test('should render correctly for a single expense' , () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={30} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render correctly for a multiple expenses' , () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={2} expenseTotal={40} />);
  expect(wrapper).toMatchSnapshot();
});
