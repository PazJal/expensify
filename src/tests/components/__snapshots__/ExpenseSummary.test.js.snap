// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly for a multiple expenses 1`] = `
<div>
  Viewing 
  2
   
  expenses
   totalling 
  $0.40
</div>
`;

exports[`should render correctly for a single expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $0.30
</div>
`;
